platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/BuildInvocationsBuilder.java

```diff
@@ -94,11 +96,13 @@ private DefaultProjectIdentifier getProjectIdentifier(Project project) {
 
     // build tasks without project reference
     private List<LaunchableGradleTask> tasks(Project project, DefaultProjectIdentifier projectIdentifier) {
-        List<LaunchableGradleTask> tasks = Lists.newArrayList();
-        for (Task task : taskLister.listProjectTasks(project)) {
-            tasks.add(buildFromTask(new LaunchableGradleTask(), projectIdentifier, task));
-        }
-        return tasks;
+        return taskLister.listProjectTasks(project).stream()
+            .map(task -> {
+                LaunchableGradleTask launchableGradleTask = buildFromTask(new LaunchableGradleTask(), projectIdentifier, task);
+                launchableGradleTask.setBuildTreePath(((TaskInternal) task).getIdentityPath().getPath());
```

Comment:
❌  This call should be moved into `buildFromTask`

I would also suggest extracting the task creation lambda into a separate method call, so that the stream mapping chain is easier to read.